core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1695,7 +1695,7 @@ public static function addImageToTemplate($objTemplate, $arrItem, $intMaxWidth=n
 		$objTemplate->picture = $picture;
 
 		// Provide an ID for single lightbox images in HTML5 (see #3742)
-		if ($strLightboxId === null && $arrItem['fullsize'])
+		if ($strLightboxId === null && $arrItem['fullsize'] && $objTemplate instanceof Template && !empty($arrItem['id']))
```

Comment:
Shouldn’t we still generate a lightbox id if it is not a template or doesn’t have an id?

Reply:
imho it's fine not to. Any custom code that uses `Controller::addImageToTemplate` probably would use the 4th parameter anyway, if they need a lightbox ID.

Reply:
Couldn't we use something that's independent of the template to generate the id?

I'm going to need this as well for #1753. :slightly_smiling_face:

Reply:
I think @ausi has a point here. If we do not generate a lightbox ID, we will end up with an empty `data-lightbox=""` attribute (e.g. see line 1755 or 1792).

Reply:
Yeah, that is the expected behavior. You only need a light box ID, if you want to group the lightboxes.

Reply:
True.

Reply:
But all lightboxes with an empty id are then also grouped - would this be OK? (compare https://github.com/contao/core/issues/3742 / comment in line 1697).

Reply:
Hm... true. But the behavior right now isn't different than before.